src/colorizer/Dataset.ts

```diff
@@ -167,13 +202,25 @@ export default class Dataset {
     this.outlierFile = manifest.outliers;
     this.tracksFile = manifest.tracks;
     this.timesFile = manifest.times;
+    this.centroidsFile = manifest.centroids;
 
     this.frames = new FrameCache(this.frameFiles.length, MAX_CACHED_FRAMES);
-    const promises = this.featureNames.map(this.loadFeature.bind(this));
-    promises.push(this.loadOutliers());
-    promises.push(this.loadTracks());
-    promises.push(this.loadTimes());
-    await Promise.all(promises);
+    const promises: Promise<unknown>[] = this.featureNames.map(this.loadFeature.bind(this));
+
+    const loadOutlier = this.loadToTexture(FeatureDataType.U8, this.outlierFile);
+    const loadTracks = this.loadToBuffer(FeatureDataType.U32, this.tracksFile);
+    const loadTimes = this.loadToBuffer(FeatureDataType.U32, this.timesFile);
+    const loadCentroids = this.loadToBuffer(FeatureDataType.U16, this.centroidsFile);
+    const loadBounds = this.loadToBuffer(FeatureDataType.U16, this.boundsFile);
+
+    promises.concat([loadOutlier, loadTracks, loadTimes, loadCentroids, loadBounds]);
+    await Promise.allSettled(promises);
+
+    this.outliers = await loadOutlier;
+    this.trackIds = await loadTracks;
+    this.times = await loadTimes;
+    this.centroids = await loadCentroids;
+    this.bounds = await loadBounds;
```

Comment:
This is much more elegant than the original! But... it looks a bit odd to have `await`s that we don't expect or want to wait for anything? (Wish I knew more about JS's async/await model to know that these definitely don't yield control, even if all these promises are definitely settled by this point. But I'm sure JS implementors in their wisdom thought of that.)

I might have tried to destructure these values out of the array returned by `Promise.allSettled`? I actually tried this to confirm doing this would typecheck properly, and I have the option to make a suggestion, so I'm curious to see what that does:

```suggestion
    const featuresPromise = Promise.allSettled(this.featureNames.map(this.loadFeature.bind(this)));

    const loadOutlier = this.loadToTexture(FeatureDataType.U8, this.outlierFile);
    const loadTracks = this.loadToBuffer(FeatureDataType.U32, this.tracksFile);
    const loadTimes = this.loadToBuffer(FeatureDataType.U32, this.timesFile);
    const loadCentroids = this.loadToBuffer(FeatureDataType.U16, this.centroidsFile);
    const loadBounds = this.loadToBuffer(FeatureDataType.U16, this.boundsFile);
    const othersPromise = Promise.all([loadOutlier, loadTracks, loadTimes, loadCentroids, loadBounds]);

    const result = await Promise.all([othersPromise, featuresPromise]);
    const [[outliers, tracks, times, centroids, bounds]] = result;

    this.outliers = outliers;
    this.trackIds = tracks;
    this.times = times;
    this.centroids = centroids;
    this.bounds = bounds;
```

I think leaving these `await`s would be just fine to be clear, but something to consider.

Reply:
Dan and I were both looking at the await statements sideways so I like this change, thank you!